sys/mips/include/atomic.h

```diff
@@ -882,6 +992,229 @@ atomic_fetchadd_64(__volatile uint64_t *p, uint64_t v)
 }
 #endif
 
+
+#ifdef __CHERI_PURE_CAPABILITY__
+/*
+ * CHERI-only variants to perform atomic operations on pointers.
+ */
+
+static __inline void
+atomic_set_ptr(volatile uintptr_t *p, uintptr_t v)
+{
+	uintptr_t temp;
+
+	__asm __volatile (
+		"1:\n\t"
+		"cllc		%0, %1\n\t"	/* load old value */
+		"cgetoffset	$t0, %0\n\t"	/* calculate new value */
```

Comment:
Shouldn't we be using the address here?

Reply:
Yes, this might have been wrong even predating the get/setaddr instructions as the base should have been taken into account.